### PR TITLE
feat(eslint): Require default argument in .reduce

### PR DESCRIPTION
**Pull request by AndesKrrrrrrrrrrr**
## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205682277164317

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -217,6 +217,16 @@
...
         ],
+        "no-restricted-syntax": [
+          "error",
+          {
+            "selector": "CallExpression[arguments.length=1] > MemberExpression.callee > Identifier.property[name='reduce']",
+            "message": "Provide initialValue to .reduce()."
+          },
+          "_FunctionExpression",
+          "_WithStatement",
+          "_BinaryExpression[operator='in']"
```
Comment (eirikurn) [inline .eslintrc.json:228]: Are these to restrict additional syntax? I don't think we want to ban function expressions.

Comment (AndesKrrrrrrrrrrr) [reply .eslintrc.json:228]: These were additional rules I thought we might want to add in the future. I put them there to easily find them later on

Comment (saevarma) [reply .eslintrc.json:228]: I think we should remove these additional rules before merging and rather bring it up on next Dev/Devops meeting if we want to restrict certain syntax in our repo.

Comment (saevarma) [reply .eslintrc.json:228]: @AndesKrrrrrrrrrrr do you want to update this PR so we can finish it?

Comment (busla) [reply .eslintrc.json:228]: added to my devops discipline meeting list

Comment (AndesKrrrrrrrrrrr) [reply .eslintrc.json]: Removed, but not discussed.

File: libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts [hunk 2]
```diff
@@ -78,17 +79,13 @@ export class Delegation extends Model<
...
-      return arrDates.reduce((a, b) => {
-        return a > b ? a : b
-      })
-    }
+      .map((x) => x.validTo) || []) as Array<Date>
 
-    return undefined
+    // Return the max value
+    const m = max(dates)
+    return m
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts:88]: ~ Redundant variable assignment.
```suggestion
    return max(dates)
```

File: libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts [hunk 3]
```diff
@@ -21,6 +21,7 @@ import { DelegationScope } from './delegation-scope.model'
 import { Domain } from '../../resources/models/domain.model'
 import { MergedDelegationDTO } from '../dto/merged-delegation.dto'
 import { DelegationType } from '../types/delegationType'
+import max from 'lodash/max'
```
Comment (saevarma) [inline libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts:24]: ~ External package dependencies should normally be above libs and relative imports. We need to look into using ESlint to force import order.

**Review by saevarma — APPROVED** [review]
Nice job! I removed `automerge` just so you can review my two minor comments then feel free to merge this.

File: libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts (hunk 2 above)
Comment (AndesKrrrrrrrrrrr) [reply libs/auth-api-lib/src/lib/delegations/models/delegation.model.ts:88]: :sweat_smile: Thanks, forgot to fix after debugging.

**Review by thorkellmani — APPROVED** [review]
LGTM 🦭

**Review by albinagu — APPROVED** [review]

**Review by davidpxn — APPROVED** [review]

**Pull request merged**